src/Polly.Shared/Duration/DurationExtensions.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Collections.Generic;
+
+namespace Polly.Duration
+{
+    /// <summary>
+    /// Extensions related to <see cref="ISleepDurationStrategy"/>.
+    /// </summary>
+    public static class DurationExtensions
+    {
+        /// <summary>
+        /// Generate a continuous sequence of <see cref="TimeSpan"/> values to use as sleep-durations.
+        /// The first <paramref name="retryCount"/> durations are generated in the same manner as
+        /// the corresponding <see cref="ISleepDurationStrategy"/>, and thereafter the maximum value 
+        /// from that sequence is returned.
+        /// For example: 2s, 4s, 6s, 8s; 8s, 8s, 8s...
+        /// </summary>
+        /// <param name="strategy">The duration strategy to use.</param>
+        /// <param name="retryCount">The maximum number of retries to use, in addition to the original call.</param>
+        public static IEnumerable<TimeSpan> Continuous(this ISleepDurationStrategy strategy, int retryCount)
```

Comment:
This method is meant to be used in the `Forever()` methods. But not sure if it's adequate.
It produces an expected series of `TimeSpan` values for the first `retryCount` iterations, and thereafter returns the `max` value it found in that sequence. 
Some of the other strategies demonstrate this better.

Reply:
Agree - I like cutting scope too; it can be painful but then you don't end up with a spaghetti api after